pysilcam/acquisition.py

```diff
@@ -97,30 +106,167 @@ def print_camera_config(camera):
+    if filename.endswith('.silc'):
+        img0 = np.load(filename, allow_pickle=False)
+    elif filename.endswith('.silc_mono'):
+        
+        img0 = np.load(filename, allow_pickle=False)
+    else:
+        img0 = imageio.imread(filename)
+    
+    # this is a quick fix to load mono 8 bit images
+    if len(np.shape(img0)) == 2:
+        img_mono =  img0
+        r, c = np.shape(img_mono)
+        img0 = np.zeros([r, c, 3], dtype=np.uint8)
+        img0[:, :, 0] = img_mono
+        img0[:, :, 1] = img_mono
+        img0[:, :, 2] = img_mono
+    
+    return img0
+
+
+def silcam_name2time(fname):
+    timestamp = pd.to_datetime(os.path.splitext(fname)[0][1:])
+    return timestamp
+
+
 class Acquire():
     '''
     Class used to acquire images from camera or disc
     '''
-    def __init__(self, USE_PYMBA=False, FAKE_PYMBA_OFFSET=0):
+    def __init__(self, USE_PYMBA=False, datapath=None, writeToDisk=False,
+                 gui=None, raw_image_queue=None,
+                 max_n_images=None, start_image_offset=0):
+        
+        self.gui = gui
+        self.raw_image_queue = raw_image_queue
+        self.datapath = datapath
+        self.max_n_images = max_n_images
+        self.start_image_offset = int(start_image_offset)
+
         if USE_PYMBA:
-            self.pymba = pymba
-            self.pymba.get_time_stamp = lambda x: pd.Timestamp.now()
-            logger.info('Pymba imported')
-            self.get_generator = self.get_generator_camera
+            self.vimba = Vimba
+            logger.info('Vimba imported')
+
+            if datapath != None:
+                os.environ['PYSILCAM_TESTDATA'] = datapath
+
+            self.writeToDisk = writeToDisk
+            self.stream_images = self.stream_from_camera
         else:
-            fakepymba.Frame.PYSILCAM_OFFSET = FAKE_PYMBA_OFFSET
-            self.pymba = fakepymba
-            logger.info('using fakepymba')
-            self.get_generator = self.get_generator_disc
+            self.stream_images = self.stream_from_disc
+            logger.info('using disc loading')
+
+    def stream_from_disc(self, config_file=None):
+        '''
+        simple disc loading version of stream_from_camera
+        wrapper for image_loader
+        used when USE_PYMBA=False
+        '''
+        logger.info('stream_from_disc')
+        disc_load_process = multiprocessing.Process(target=self.image_loader)
+        disc_load_process.start()
+        logger.debug(f"disc_load_process started. Name: {disc_load_process.name}")
+        # while True:
+        #     try:
+        #         time.sleep(10000)
+        #     except KeyboardInterrupt:
+        #         logger.info('User interrupt with ctrl+c, terminating PySilCam.')
+        #         break
+
+        # disc_load_process.join()
+        logger.debug("Exiting reading from disc threads.")
+        return disc_load_process
+        # sys.exit(0)
+
+    def image_loader(self):
```

Comment:
C901: 'Acquire.image_loader' is too complex (11)  <!-- Automatically posted by Lintly -->